src/Test.UnitTests.Sarif/Writers/BaseLoggerTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Writers
+{
+    public class BaseLoggerTests
+    {
+        [Fact]
+        public void BaseLogger_ShouldAcceptEmptyAndNullKindsAndFailures()
+        {
+            //  null levels, null kinds
+            BaseLoggerTestConcrete baseLoggerTestConcrete = new BaseLoggerTestConcrete(null, null);
+            AssertLevelAndKindAreNonEmpty(baseLoggerTestConcrete);
+
+            //  null levels, empty kinds
+            baseLoggerTestConcrete = new BaseLoggerTestConcrete(null, new List<ResultKind> { });
+            AssertLevelAndKindAreNonEmpty(baseLoggerTestConcrete);
+
+            //  empty levels, null kinds
+            baseLoggerTestConcrete = new BaseLoggerTestConcrete(new List<FailureLevel> { }, null);
+            AssertLevelAndKindAreNonEmpty(baseLoggerTestConcrete);
+
+            //  empty levels, empty kinds
+            baseLoggerTestConcrete = new BaseLoggerTestConcrete(new List<FailureLevel> { }, new List<ResultKind> { });
+            AssertLevelAndKindAreNonEmpty(baseLoggerTestConcrete);
+        }
+
+        private void AssertLevelAndKindAreNonEmpty(BaseLoggerTestConcrete baseLoggerTestConcrete)
+        {
+            Assert.True(baseLoggerTestConcrete.FailureLevelsPublicViewer.Count > 0);
+            Assert.True(baseLoggerTestConcrete.ResultKindPublicViewer.Count > 0);
+        }
+
+        [Fact]
+        public void BaseLogger_ShouldCorrectlyDefault()
+        {
+            BaseLoggerTestConcrete baseLoggerTestConcrete = new BaseLoggerTestConcrete(null, null);
+            
+            Assert.True(baseLoggerTestConcrete.ResultKindPublicViewer.Count == 1);
```

Comment:
>Count == 1 [](start = 70, length = 10)

.Count.Should().Be(1)

Reply:
Done.

---
In reply to: [560613522](https://github.com/microsoft/sarif-sdk/pull/2249#discussion_r560613522) [](ancestors = 560613522)